### PR TITLE
MOVE ONLY: move rpc* to rpc/

### PR DESCRIPTION
**Pull request by dcousens**
Was just playing around with the RPC,  and found the constant prefixing of file names frustrating.

This also seems to be in line with the way other modules (`zmq` etc) have been grouped.

Should be easy to review.

**Note:** This won't negatively impact git blame

### CODE REVIEW

File: src/bitcoind.cpp [hunk 1]
```diff
@@ -5,14 +5,13 @@
...
 #include "clientversion.h"
-#include "rpcserver.h"
+#include "rpc/server.h"
 #include "init.h"
 #include "noui.h"
 #include "scheduler.h"
 #include "util.h"
 #include "httpserver.h"
 #include "httprpc.h"
-#include "rpcserver.h"
```
Comment (dcousens) [inline src/bitcoind.cpp]: Not sure why there was two `include`s here, but I removed the latter without ill effect.  I can put it back if necessary.

**Comment by MarcoFalke** [comment]
What's up with `src/wallet/rpc*`?

> just renamed, not moved

I don't think there is a difference to git.

File: src/rpc/server.h [hunk 2]
```diff
@@ -198,7 +198,7 @@ extern UniValue estimatepriority(const UniValue& params, bool fHelp);
 extern UniValue estimatesmartfee(const UniValue& params, bool fHelp);
 extern UniValue estimatesmartpriority(const UniValue& params, bool fHelp);
 
-extern UniValue getnewaddress(const UniValue& params, bool fHelp); // in rpcwallet.cpp
+extern UniValue getnewaddress(const UniValue& params, bool fHelp); // in rpc/wallet.cpp
```
Comment (MarcoFalke) [inline src/rpc/server.h]: There is no `rpc/wallet.cpp`?!

**Comment by jonasschnelli** [comment]
Nice!
Concept ACK.

File: src/rpc/server.h (hunk 2 above)
Comment (dcousens) [inline src/rpc/server.h]: Indeed.
I wonder what the best course of action is here?
I'll probably just change that to `wallet/rpcwallet.cpp`

**Comment by dcousens** [comment]
> I don't think there is a difference to git.

If it doesn't pick up the re-name,  the diff would have been much larger.

File: src/rpc/server.h (hunk 2 above)
Comment (jonasschnelli) [inline src/rpc/server.h]: #7307 would solve that.
But I agree, changing to `wallet/rpcwallet.cpp` is probably fine for this PR.

**Comment by MarcoFalke** [comment]
Git tracks content, not files. You will lose the history of those files but not the authorship (git blame), iiuc.

**Comment by dcousens** [comment]
@MarcoFalke I was just trying to differentiate from how a MOVE operation can look like [here](https://github.com/bitcoin/bitcoin/pull/7311).
I'd rather not fill this PR with bike shedding about irrelevant things though,  so I'll probably delete this post [and the others related to this] in a few hours.

**Comment by MarcoFalke** [comment]
@dcousens Somehow had the impression this was IRC. ^^

**Comment by laanwj** [comment]
Concept ACK. 

Not that you moved it, but for completeness' sake: please leave rpcwallet.cpp in src/wallet. e.g. it is the RPC interface to the wallet. This, imo, logically belongs to the wallet: if there was a different wallet it could have a different RPC interface.

File: src/rpc/server.h (hunk 2 above)
Comment (laanwj) [inline src/rpc/server.h]: Yes, my suggested fix too would b to just fix the comment

**Comment by dcousens** [comment]
@laanwj amended inline comment commit.  Should be good to go.

**Comment by MarcoFalke** [comment]
@dcousens Could you make the commits atomic, so that each of them compiles?

**Comment by dcousens** [comment]
@MarcoFalke rebased and fixed a build error I hadn't seen in `qt/` (I don't build with Qt).
Commits were already atomic.

**Comment by MarcoFalke** [comment]
Concept ACK 611508c

**Comment by laanwj** [comment]
Needs rebase after #7307

**Comment by dcousens** [comment]
Rebased

**Comment by MarcoFalke** [comment]
utACK d13f65ebac13ec18b7eb55176c31f1404f185c0c

**Comment by paveljanik** [comment]
ACK https://github.com/dcousens/bitcoin/commit/d13f65ebac13ec18b7eb55176c31f1404f185c0c

Header guards do not match now (`_` missing), but not that important.

**Comment by jonasschnelli** [comment]
utACK d13f65ebac13ec18b7eb55176c31f1404f185c0c.
Confirm move only.

**Comment by jtimon** [comment]
nice

**Pull request merged**